packages/dds/tree/src/test/utils.ts

```diff
@@ -113,12 +113,15 @@ export class TestTreeProvider {
      * Give this {@link TestTreeProvider} the ability to summarize on demand during a test by creating a summarizer
      * client for the container at the given index. This must be called before any trees submit any edits, or else a
      * different summarizer client might already have been elected.
-     * @param index - the container that will spawn the summarizer client
      * @returns a function which will cause a summary to happen when awaited. May be called multiple times.
      */
-    public async enableManualSummarization(index = 0): Promise<() => Promise<void>> {
-        assert(index < this.trees.length, "Index out of bounds: not enough trees");
-        const summarizer = await createSummarizer(this.provider, this.containers[index]);
+    public async enableManualSummarization(): Promise<() => Promise<void>> {
+        // TODO: support manual summarization with multiple trees once bug #1876 is fixed.
```

Comment:
Issue #1876 is already closed.

If instead you mean https://dev.azure.com/fluidframework/internal/_workitems/edit/1876 in our internal tracker thats a problem since it is not accessible to many people who might look at this.

Also, I'm not sure thats the right item to reference here or that item's status is wrong. Its status indicates that it's in review (meaning that someone, presumable you, has a fix for that bug in PR). If that really was the case, we should probably wait for that bug to be fixed (and thus manual summarization with multiple trees would work) and not do this PR at all.

If we are going to work around a bug without fixing it, we should not track the workaround as if it's fixing the bug: it should get its own work item.

If we are going to leave a known bug in the codebase, it should be documented where all users of the codebase can see, meaning on github (as an issue and/or documentation in the codebase) and not just our internal tracker.

Reply:
If we are going to require "This may only be called when exactly one tree has been created and has not submitted any edits", I'd like to code to document why this is the case. If its a bug we can explain that. If its not a bug, we should still have some explanation. Avoiding the whole issue as @noencke proposes by moving this to an option of create sounds like a great idea.

Reply:
And thanks for the details @noencke : that helps a lot!

Reply:
> and you assert that if `summarizeOnDemand` is true then `trees === 1`

Wouldn't it be possible to support `trees !== 0` instead? You'd create the first tree, then make it the summarizer, then create the rest of the trees (if any).